### PR TITLE
Display mentor details during Mentee application period

### LINKED ISSUE #276: Display mentors and their details publicly during Mentee Application Period
**Is your feature request related to a problem? Please describe.**
When the program is in the MENTEE_APPLICATION period, users should be able to see the mentors by selecting a program from the homepage. The design of the page is already done.

**Describe the solution you'd like**
- Get the approved mentors using the related endpoints. (`GET api/programs/{id}/mentors?states='APPROVED'` or any other way)
- Display the mentors in the table
- When the user clicks on a mentor name, show the mentor details
- If the user clicks the `<- ` (back button) on the **mentor details** page, navigate back to the **mentors** page

**Describe alternatives you've considered**
N/A

**Additional context**
PR for Design: https://github.com/sef-global/scholarx-frontend/pull/256

### PR DESCRIPTION
**Pull request by Piumal1999**
## Purpose

The purpose of this PR is to fix #276 

## Goals

Display mentors and their details publicly during Mentee Application period

## Approach

Replaced dummy data with actual data got from the backend
removed unused variables and methods

## Screenshots

https://user-images.githubusercontent.com/27498587/147689360-d46eb978-c770-4cba-a2bb-735e052c7f6b.mov

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Related PRs
 
#256

## Test environment
 
MacOs

## Learning

N/A

### CODE REVIEW

File: src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx [hunk 1]
```diff
@@ -119,12 +56,39 @@ function Mentors() {
...
-          pagination={false}
-        />
+        <Table dataSource={mentors} rowKey="id">
+          <Column
+            title="Mentor"
+            dataIndex={''}
+            render={(mentor: Mentor) => (
+              <a
+                href={
+                  '/program/' + programId + '/mentor/' + mentor.id + '/view'
```
Comment (Gravewalker666) [inline src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx]:
```suggestion
                  `/program/${programId}/mentor/${mentor.id}/view`
```

Comment (Gravewalker666) [reply src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx]: And having an anchor tag would refresh the entire page, We can use ant designs `Link` component to direct the users without refreshing the page

**Review by Gravewalker666 — CHANGES_REQUESTED** [review]
Rest of the things looks fine, we could improve the UI afterwards 
Requested some minor changes

File: src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx (hunk 1 above)
Comment (Piumal1999) [reply src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx]: Ah right. fixed with a5abb9c

Comment (Gravewalker666) [reply src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx]: Nice

**Review by Gravewalker666 — APPROVED** [review]
Good job @Piumal1999

**Pull request merged**